applications/CoSimulationApplication/custom_python/add_co_sim_io_to_python.cpp

```diff
@@ -0,0 +1,478 @@
+// KRATOS  / ___|___/ ___|(_)_ __ ___  _   _| | __ _| |_(_) ___  _ ___
+//        | |   / _ \___ \| | '_ ` _ \| | | | |/ _` | __| |/ _ \| '_  |
+//        | |__| (_) |__) | | | | | | | |_| | | (_| | |_| | (_) | | | |
+//         \____\___/____/|_|_| |_| |_|\__,_|_|\__,_|\__|_|\___/|_| |_|
+//
+//  License:		 BSD License
+//					 license: CoSimulationApplication/license.txt
+//
+//  Main authors:    Philipp Bucher (https://github.com/philbucher)
+//
+
+// System includes
+#include <unordered_map>
+#include <map>
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "includes/model_part.h"
+#include "custom_python/add_co_sim_io_to_python.h"
+
+// CoSimIO
+// defining the macros that are used within the CoSimIO to point the the Kratos-native macros
+// Note that this has to be done BEFORE including the header the first time
+#define KRATOS_CO_SIM_INFO KRATOS_INFO
+#define KRATOS_CO_SIM_INFO_IF KRATOS_INFO_IF
+#define KRATOS_CO_SIM_ERROR KRATOS_ERROR
+#define KRATOS_CO_SIM_ERROR_IF KRATOS_ERROR_IF
+#define KRATOS_CO_SIM_ERROR_IF_NOT KRATOS_ERROR_IF_NOT
+#include "co_sim_io/co_sim_io.h"
```

Comment:
Macros part I

@pooyan-dadvand we already discussed the macro-issue, I think I solved it now, see here

Reply:
see this file:
https://github.com/KratosMultiphysics/Kratos/blob/cosim/cosim_io/applications/CoSimulationApplication/co_sim_io/impl/co_sim_io_macros.h

If the macros are not predefined (which I am doing here, since we directly want to use the Kratos macros) then it uses its own
I don't know TBH though how this could work for other codes. I basically did what you told me (or at least what I understood)